Update live code to live demos in 17-live.md (#1558)
carpentries/instructor-training: episodes/17-live.md

---
start: no
title: Live Demonstration is a Skill
block: Building Teaching Skill
teaching: 20
exercises: 45
---

One of the cornerstones of The Carpentries teaching is live
demonstrations: *instructors do not use slides to teach*, but work through the lesson material,
typing in code or modeling procedures, with the workshop participants following
along. This section explains how it works, why we use it, and

> gives general tips for an effective participatory live demo presentation. We will

Reviewer: ```suggestion
gives general tips for an effective participatory live demonstration. We will
```
"demo presentation" reads funny to me

:::::::::::::::::::::::::::::::::::::::: questions

- Why do we teach using participatory live demos?

::::::::::::::::::::::::::::::::::::::::::::::::::

## Why Participatory Live Demos?

We do not use slides in our lessons.
Instead, instructors share their computer screens
and work through the lesson,
typing in the code,
reformatting data,
demonstrating commands and procedures,
and talking as we go.
This is called "live coding" or more generally "live demonstration".
However, the instructor is not live coding in a vacuum.
Importantly, learners are strongly encouraged
to "code-along" with the instructor.
We refer to the practice of having the instructor live code and the learners code along as "participatory live coding" or, less formally, 'code-along sessions'.

:::::::::::::::::::::::::::::::::::::::::  callout

## Live Coding or Live Demonstration?

The term "live coding" is used throughout instructor training
to describe the Carpentries model of teaching,
but you don't need to be a programmer or know how to code at all
to integrate live demonstrations in your teaching.
Many of our lessons on tools like spreadsheets and OpenRefine

> include no coding at all, and similar models are widely used

Reviewer 2: This is not entirely correct, although I see the distinction of these two lessons. Both use some use of functions. Might change it to "Some lessons might not include coding..."

::::::::::::::::::::::::::::::::::::::::::::::::::

> Live demos fits well into the practice-feedback model we have been discussing - by providing

Reviewer: ```suggestion
Live demonstration fits well into the practice-feedback model we have been discussing - by providing
```
i think using the full work is better for non native english speakers, but would accept the singular of the shortcut

learners with continuous opportunities for practice (every time they type a line of code or perform a procedure)

> and continuous feedback (their attempt either works or fails with an error message). It is

Reviewer: the "error message" part may not translate to non-coding contexts, but i'm not sure what the best fix is here.

Reviewer 3: Error messages defnitely come up in non-coding contexts IME.

## Compare and Contrast

> Watch this first participatory live demo video: [https://youtu.be/bXxBeNkKmJE][live-coding-bad]

Reviewer: this is actually code, so I think here, live "coding" is actually more accurate and we should augment the exercise with reflection on what we can learn from this that applies to non-coding contexts

:::::::::::::::::::::::::::::::::::::::: keypoints

> - Live demos forces the instructor to slow down.

Reviewer: ```suggestion
- Live demonstration forces the instructor to slow down.
```
or live demo maybe, but "live demos" feels like a mismatch of plurality?
